### ISSUE TITLE
Feature Request:  Indicate Fish Resources in Water

### ISSUE THREAD

**Issue by akuenzi** [Return-To-The-Roots/s25edit#28]
When working in the map editor, it would be interesting if one could specify the amount of 'fish resources' on water bodies, similar to how one specifies coal, gold, iron ore, or granite resources.  This would then dictate how many fish the fisherman could catch in-game before that particular area of water is depleted.

It would be helpful if there was also an 'unlimited' tag for fish (or even mining) resources, or a resource that does not get depleted no matter how much is harvested.

Thank you for considering this! :)

**Comment by Spikeone** [comment]
This would also need a change in RttR and it would break the settlers2 map format. 

https://settlers2.net/documentation/world-map-file-format-wldswd/

Fish is only "there" or it's not, but there is no amount on that. So adding fish amount would change the mapformat and render it incompatible to settlers2.
Also RttR would need to understand, that there is not fish, but an amount of fish in water.

**Comment by Flamefire** [comment]
Correct. No change in map format is planned as there are still players wanting to play with original S2 (or the new one) so this cannot be done.

IIRC there is an addon in RttR which has exhaustible fish which is possible without changing the map format.

**Comment by Spikeone** [comment]
I'd not close this one yet - we always wanted to add a RttR Map format, some day, where you can have prebuilt buildings etc., thats a feature we could add then, can't we?

**Comment by Flamefire** [comment]
Realistically that's not happening. There are ~230 issues open in the main repo and the number doesn't really decrease. 

However I've just seen that this is indeed possible. We have an addon "InexhaustibleFish" and only use "4" as the default amount of fish. The map format seems to allow for differing fish amounts but I'm not sure if S2 can deal with that. It seems to always use `0x87` for fish which is actually "7 fish" (0x80+7) but we treat it as 4 fish for some reason (maybe because the original did it that way)

So IFF S2 can deal with other values than `0x87` and treats it correctly (either ignoring or handling the amount) then we could do this via addon and a change to the editor. But the problem then is: How to show the amount of fish? The only reasonable answer seems to be: Don't.

**Comment by Spikeone** [comment]
We could simply add a stack, like for resources?

Also we can simply mark the map as invalid, so it doesn't show up in settlers2 - but then again, how should people know about it, if they don't know the details of how everything worked. Maybe a message "This map is incompatible to Settlers2 due to xy" on saving?
